src/inmanta/graphql/schema.py

```diff
@@ -101,14 +61,60 @@ def add_filter_and_sort(
     return stmt
 
 
-def initialize_schema() -> strawberry.Schema:
+def get_schema(metadata: GraphQLMetadata) -> strawberry.Schema:
     """
     Initializes the Strawberry GraphQL schema.
     It is initiated in a function instead of being declared at the module level, because we have to do this
     after the SQLAlchemy engine is initialized.
     """
+
+    mapper: StrawberrySQLAlchemyMapper[typing.Any] = StrawberrySQLAlchemyMapper()
     loader = StrawberrySQLAlchemyLoader(async_bind_factory=get_session_factory())
 
+    def get_expert_mode(root: "Environment") -> bool:
```

Comment:
With the string-based type annotation it's rather hard to see which environment type is intended. We have multiple classes called Environment. I think it would help to make this not a string-based type.

Reply:
But this has to be defined before `Environment`. What do you suggest to use instead?

Reply:
I see. I don't think there is much we an do about that.